### PR TITLE
#15 (page_action icon light/dark added), #110 and few improvements GUI

### PR DESCRIPTION
**Pull request by Sav22999**
- Issue #111
- Some improvements GUI: checkbox now use "browser-style", it's more beautiful.

Options now looks:
immagine

### CODE REVIEW

**Comment by Sav22999** [comment]
I don't understand ...

File: manifest.json [hunk 1]
```diff
@@ -5,41 +5,41 @@
...
+            "16": "icon.svg",
+            "24": "icon.svg",
+            "48": "icon.svg",
+            "96": "icon.svg"
         },
         "theme_icons": [{
-            "dark": "icon-dark.svg",
-            "light": "icon-light.svg",
+            "dark": "icon.svg",
+            "light": "z.svg",
```
Comment (Mte90) [inline manifest.json]: this file doesn't exist.
Please revert everything as the original manifest and filesnames

File: options/options.js [hunk 2]
```diff
@@ -144,3 +160,53 @@ function updateState() {
...
+  // move the element up
+  var id_temp = (id.toString()).replace("-move-up","");
+  document.getElementById(id_temp+"-priority").value = parseInt(document.getElementById(id_temp+"-priority").value)-1;
+  // call "onchange" event (forced)
+  if ("createEvent" in document) {
+    var evt = document.createEvent("HTMLEvents");
+    evt.initEvent("change", false, true);
+    document.getElementById(id_temp+"-priority").dispatchEvent(evt);
+  }
+  else
```
Comment (Mte90) [inline options/options.js]: please be consistent with style add the `{}`

File: options/options.js [hunk 3]
```diff
@@ -144,3 +160,53 @@ function updateState() {
...
+    document.getElementById(id_temp+"-priority").fireEvent("onchange");
+  setFocusRowTemp(parseInt(document.getElementById(id_temp+"-priority").value)-1);
+  document.getElementById(id_temp+"-priority").focus();
+}
+
+function moveDown(id) {
+  // move the element down
+  var id_temp = (id.toString()).replace("-move-down","");
+  document.getElementById(id_temp+"-priority").value = parseInt(document.getElementById(id_temp+"-priority").value)+1;
```
Comment (Mte90) [inline options/options.js]: this code is a duplicate with the above method better create another function to simplify the maintenance.

File: options/options.js [hunk 4]
```diff
@@ -144,3 +160,53 @@ function updateState() {
...
+  setFocusRowTemp(parseInt(document.getElementById(id_temp+"-priority").value)-1);
+  document.getElementById(id_temp+"-priority").focus();
+}
+
+function setFocusRowTemp(index) {
+  // set "focus" on row modified, before clear eventually wrong "class"
+  clearFocusRowTemp();
+  document.getElementsByClassName("service")[index].classList.add('row_focus');
+  setTimeout(function() {
+    document.getElementsByClassName("service")[index].classList.remove('row_focus')
```
Comment (Mte90) [inline options/options.js]:
```
service = document.getElementsByClassName("service")[index];
service.classList.add('row_focus');
setTimeout(function() {
    service.classList.remove('row_focus')
```

File: options/options.js [hunk 5]
```diff
@@ -144,3 +160,53 @@ function updateState() {
...
+  }, 1000);
+}
+
+function clearFocusRowTemp()
+{
+  document.querySelectorAll('div').forEach(function(item) {
+    if (item.classList.contains("service")) {
```
Comment (Mte90) [inline options/options.js]: I don't think that this if is required, if the method is clear we want to remove the class to any item.

File: options/options.js (hunk 3 above)
Comment (Mte90) [reply options/options.js]: Also associate `document.getElementById(id_temp+"-priority")` to a var for performance and readibility

File: options/options.js [hunk 6]
```diff
@@ -19,6 +19,22 @@ function restoreOptions() {
...
+      });
+      document.getElementById(item.id).innerHTML="▲";
+    }
+    else if(item.id.endsWith("move-down")) {
+      document.getElementById(item.id).addEventListener('click', function () {
+        moveDown(item.id);
+      });
+      document.getElementById(item.id).innerHTML="▼";
+    }
+  });
```
Comment (Mte90) [inline options/options.js]:
```  
document.querySelectorAll('button[id]').forEach(function(item) {
document.getElementById(item.id).addEventListener('click', function () {
    if (this.endsWith("move-up")) {
        moveUp(item.id);
      this.innerHTML="▲";
    } else if(this.endsWith("move-down")) {
        moveDown(item.id);
      this.innerHTML="▼";
    }
  });
});
```

Comment (Mte90) [reply options/options.js]: less core more easy to mantain :-)

Comment (Sav22999) [reply options/options.js]: Absolutely ok 👍 I'll change it as soon as

File: options/options.js (hunk 5 above)
Comment (Sav22999) [reply options/options.js]: It cause some bugs ... if you press fast on some buttons (differents) -> they stay "focussed"

File: manifest.json (hunk 1 above)
Comment (Sav22999) [reply manifest.json]: Oh, it's a typo. I'm so sorry

File: options/options.js (hunk 6 above)
Comment (Sav22999) [reply options/options.js]: This doesn't work

**Comment by Sav22999** [comment]
I was wrong, the last commit fixed "#111" and not "#110"

File: background.js [hunk 7]
```diff
@@ -37,28 +40,56 @@ browser.tabs.onUpdated.addListener(function(tabId, changeInfo, tab) {
...
-          popup: '/modal/modal.html'
-        });
-      }
+setPageActionIcon("light")
+
+function getStyle(themeInfo)
+{
+  var rgb_list = ["rgb(71, 71, 73)", "rgb(50, 50, 52)"]; // list of RGB where icon_theme should be light
+  if (themeInfo.colors)
+  {
```
Comment (Mte90) [inline background.js]: keep the code consistent with braces on the condition line

**Comment by Sav22999** [comment]
Do you mean:
"function ... {
...
}"?

**Comment by Mte90** [comment]
I mean `if (themeInfo.colors) {`not like it is now.

File: options/options.js [hunk 8]
```diff
@@ -19,6 +19,22 @@ function restoreOptions() {
...
+      document.getElementById(item.id).addEventListener('click', function () {
+        moveUpDown(item.id, "up");
+      });
+      document.getElementById(item.id).innerHTML="▲";
+    }
+    else if(item.id.endsWith("move-down")) {
+      document.getElementById(item.id).addEventListener('click', function () {
+        moveUpDown(item.id, "down");
+      });
+      document.getElementById(item.id).innerHTML="▼";
```
Comment (Mte90) [inline options/options.js]: Why add that symbol with js instead of HTML?

Comment (Mte90) [inline options/options.js]:
```
document.querySelectorAll('button[id]').forEach(function(item) {
    document.getElementById(item.id).addEventListener('click', function () {
      if (item.id.endsWith("move-up")) {
        moveUpDown(item.id, "up");
        return;
      } 
      moveUpDown(item.id, "down");
    });
  });
```
With that change this now works

**Comment by Mte90** [comment]
About the icon change this works only with the default Firefox themes and when the theme is changed require a restart of Firefox to load the changes.

File: options/options.js (hunk 8 above)
Comment (Sav22999) [reply options/options.js]: So you can change the symbol fast and easilier (maybe also remove it and replace it with image).

**Comment by Sav22999** [comment]
Yes, I know... maybe we should inform of it?

File: options/options.js (hunk 8 above)
Comment (Mte90) [reply options/options.js]: Yes but the button doesn't move and there are two, so is better to define them as default and simplify the code.
Probably is more easy to use css with `content` attribute to edit the html.

Comment (Sav22999) [reply options/options.js]: As you prefer 😄 I wasn't sure about that symbol, so I would like your opinion. 👍

**Comment by Sav22999** [comment]
@Mte90 so, about the icon, what should I do?

**Pull request merged**